pkg/agent/controller/egress/egress_controller.go

```diff
@@ -1069,7 +1067,7 @@ func (c *EgressController) syncEgress(egressName string) error {
 			return err
 		}
 		if assigned {
-			c.record.Eventf(egress, corev1.EventTypeNormal, "IPAssigned", "Assigned Egress %s with IP %s on Node %s", egress.Name, desiredEgressIP, desiredNode)
+			c.record.Eventf(egress, "", corev1.EventTypeNormal, "IPAssigned", "NodeAssignment", "Assigned Egress %s with IP %s on Node %s", egress.Name, desiredEgressIP, desiredNode)
```

Comment:
The second argument seems wrong (https://pkg.go.dev/k8s.io/client-go@v0.35.0/tools/internal/events#EventRecorder). Shouldn't it be `nil`. Did you run `make golangci` locally?

I will run the workflows, but I expect it to fail